### PR TITLE
Fix bugs found in PE-D (1 - Exclude EditTask)

### PR DESCRIPTION
**Pull request by dannytayjy**
Here are the bug fixes to the issues found in PE-D.
This PR excludes bug fixes to EditTask.

#192 - Changed to the correct ParseException error message for index that is a non-zero unsigned integer

Issue 192

#202 - Added an extra condition to check if START_TIME and END_TIME are the same, as the advertised behaviour stated in UG is that START_TIME must be earlier than END_TIME

Issue 202

#204 - Changed to the correct ParseException error message to reflect that event END_TIME cannot be earlier than the START_TIME 

Issue 204

#211 - Changed the error message of incorrect time format to better reflect the acceptable range of time input
#223 - Changed "HHMM" in the error message of incorrect time format to "HHmm" to standardise with UG

Issue 211   223

#219 - Changed the error message to better reflect on how the tagTask feature works, which is only able to tag the employee to a task via the employee's full name only, as there may be more than one employee with the same partial name

Issue 219_01

Issue 219_02

#222 & #231 - Changed the error message to better reflect on how the untagTask feature works, which is only able to untag the employee from a task via the employee's full name only, as there may be more than one employee with the same partial name

Issue 222

Issue 231

#193 - Optimize GUI and set minimum width and height to ensure that the user is not able to resize the application window until it is too small to see both the employees list and the tasks list

Issue 193

### CODE REVIEW

File: src/main/java/manageezpz/commons/core/GuiSettings.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 public class GuiSettings implements Serializable {
 
     private static final double DEFAULT_HEIGHT = 600;
-    private static final double DEFAULT_WIDTH = 740;
+    private static final double DEFAULT_WIDTH = 850;
```
Comment (Denniszedead) [inline src/main/java/manageezpz/commons/core/GuiSettings.java:14]: Good job preventing any magical variables!

File: src/main/java/manageezpz/logic/commands/AddEventTaskCommand.java [hunk 2]
```diff
@@ -17,12 +17,12 @@
             + ": Adds an Event Task to ManageEZPZ.\n"
             + "Parameters: "
             + PREFIX_DESCRIPTION + "DESCRIPTION "
-            + PREFIX_AT_DATETIME + "[DATE] [START_TIME] [END_TIME]\n"
+            + PREFIX_AT_DATETIME + "DATE START_TIME END_TIME\n"
```
Comment (ChanWeiJie) [inline src/main/java/manageezpz/logic/commands/AddEventTaskCommand.java:20]: Thanks for removing the "[ ]"!

File: src/main/java/manageezpz/logic/commands/EditTaskCommand.java [hunk 3]
```diff
@@ -165,7 +173,7 @@ private Task updateEvent(Event currentTask, String desc, String date, String tim
             Time newStartTime = ParserUtil.parseTime(newStartEndTimeStrParts[0]);
             Time newEndTime = ParserUtil.parseTime(newStartEndTimeStrParts[1]);
 
-            if (newEndTime.getParsedTime().compareTo(newStartTime.getParsedTime()) < 0) {
+            if (newEndTime.getParsedTime().compareTo(newStartTime.getParsedTime()) < 1) {
```
Comment (Denniszedead) [inline src/main/java/manageezpz/logic/commands/EditTaskCommand.java:176]: May i know what the 1 means?

File: src/main/java/manageezpz/logic/parser/EditTaskCommandParser.java [hunk 4]
```diff
@@ -23,25 +23,29 @@
      */
     @Override
     public EditTaskCommand parse(String args) throws ParseException {
-        requireNonNull(args);
-
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_DESCRIPTION, PREFIX_AT_DATETIME, PREFIX_DATE);
 
+        System.out.println(argMultimap.getPreamble());
```
Comment (ChanWeiJie) [inline src/main/java/manageezpz/logic/parser/EditTaskCommandParser.java:29]: Do remember to remove any unused print statements!

File: src/main/java/manageezpz/logic/commands/EditTaskCommand.java (hunk 3 above)
Comment (dannytayjy) [reply src/main/java/manageezpz/logic/commands/EditTaskCommand.java:176]: This is to factor in the condition that the newEndTime cannot be the same as the newStartTime.

File: src/main/java/manageezpz/logic/commands/EditTaskCommand.java [hunk 5]
```diff
@@ -72,6 +77,7 @@
      * @param time New time of the Task
      */
     public EditTaskCommand(Index index, String desc, String date, String time) {
+        requireAllNonNull(index, desc, date, time);
```
Comment (aaron-ljx) [inline src/main/java/manageezpz/logic/commands/EditTaskCommand.java:80]: Good job on ensuring that the parameters that are passed in are not null.

**Review by ChanWeiJie — APPROVED** [review]
LGTM!

File: src/main/java/manageezpz/logic/commands/EditTaskCommand.java (hunk 3 above)
Comment (Denniszedead) [reply src/main/java/manageezpz/logic/commands/EditTaskCommand.java:176]: Is it possible to abstract it into a method?

**Pull request merged**